test/conformance/helpers/broker_data_plane_test_helper.go

```diff
@@ -0,0 +1,458 @@
+) {
+	channelTestRunner.RunTests(t, testlib.FeatureBasic, func(t *testing.T, channel metav1.TypeMeta) {
+		client := testlib.Setup(t, true, setupClient...)
+		defer testlib.TearDown(client)
+		brokerName := "br"
+		triggerName := "trigger"
+		loggerName := "logger-pod"
+		eventTracker, _ := recordevents.StartEventRecordOrFail(client, loggerName)
+		defer eventTracker.Cleanup()
+		client.WaitForAllTestResourcesReadyOrFail()
+
+		config := client.CreateBrokerConfigMapOrFail(brokerName, &channel)
+
+		broker := client.CreateBrokerV1Beta1OrFail(
+			brokerName,
+			resources.WithBrokerClassForBrokerV1Beta1(brokerClass),
+			resources.WithConfigForBrokerV1Beta1(config),
+		)
+		client.WaitForResourceReadyOrFail(broker.Name, testlib.BrokerTypeMeta)
+
+		trigger := client.CreateTriggerOrFailV1Beta1(
+			triggerName,
+			resources.WithBrokerV1Beta1(broker.Name),
+			resources.WithAttributesTriggerFilterV1Beta1(eventingv1beta1.TriggerAnyFilter, eventingv1beta1.TriggerAnyFilter, nil),
+			resources.WithSubscriberServiceRefForTriggerV1Beta1(loggerName),
+		)
+
+		client.WaitForResourceReadyOrFail(trigger.Name, testlib.TriggerTypeMeta)
+
+		t.Run("Ingress Supports CE0.3", func(t *testing.T) {
+			eventID := "CE0.3"
+			event := ce.NewEvent()
+			event.SetID(eventID)
+			event.SetType(testlib.DefaultEventType)
+			event.SetSource("CE0.3")
+			body := fmt.Sprintf(`{"msg":%q}`, eventID)
+
+			if err := event.SetData(ce.ApplicationJSON, []byte(body)); err != nil {
+				t.Fatalf("Cannot set the payload of the event: %s", err.Error())
+			}
+			event.Context.AsV03()
+			event.SetSpecVersion("0.3")
+			client.SendEventToAddressable("v03-test-sender", broker.Name, testlib.BrokerTypeMeta, event)
+			originalEventMatcher := recordevents.MatchEvent(cetest.AllOf(
+				cetest.HasSource(event.Source()),
+				cetest.HasSpecVersion("0.3"),
+			))
+			eventTracker.AssertAtLeast(1, originalEventMatcher)
+		})
+
+		t.Run("Ingress Supports CE1.0", func(t *testing.T) {
+			eventID := "CE1.0"
+			event := ce.NewEvent()
+			event.SetID(eventID)
+			event.SetType(testlib.DefaultEventType)
+			event.SetSource("CE1.0")
+			body := fmt.Sprintf(`{"msg":%q}`, eventID)
+			if err := event.SetData(ce.ApplicationJSON, []byte(body)); err != nil {
+				t.Fatalf("Cannot set the payload of the event: %s", err.Error())
+			}
+
+			client.SendEventToAddressable("v10-test-sender", broker.Name, testlib.BrokerTypeMeta, event)
+			originalEventMatcher := recordevents.MatchEvent(cetest.AllOf(
+				cetest.HasSource(event.Source()),
+			))
+			eventTracker.AssertAtLeast(1, originalEventMatcher)
+
+		})
+		t.Run("Ingress Supports Structured Mode", func(t *testing.T) {
+			eventID := "Structured-Mode"
+			event := ce.NewEvent()
+			event.SetID(eventID)
+			event.SetType(testlib.DefaultEventType)
+			event.SetSource("CEStructured Mode")
+			body := fmt.Sprintf(`{"msg":%q}`, eventID)
+			if err := event.SetData(ce.EncodingStructured.String(), []byte(body)); err != nil {
```

Comment:
You're not sending in structured mode this way, you're just setting the content type of the data (and i'm not sure the value you're setting is legal)